src/components/duck.js

```diff
@@ -0,0 +1,53 @@
+AFRAME.registerComponent("duck", {
+  schema: {
+    initialForce: { default: 0 },
+    maxForce: { default: 6.5 },
+    maxScale: { default: 5 }
+  },
+
+  init: function() {
+    this.system = this.el.sceneEl.systems.physics;
+    this.hasBody = false;
+    this.position = new THREE.Vector3();
+    this.force = new THREE.Vector3(0, this.data.initialForce, 0);
+    this.initialScale = this.el.object3D.scale.x;
+    this.maxScale = this.data.maxScale * this.initialScale;
+
+    this._handleGrabStart = this._handleGrabStart.bind(this);
+  },
+
+  play: function() {
+    this.system.addComponent(this);
+    this.el.addEventListener("grab-start", this._handleGrabStart);
+  },
+
+  pause: function() {
+    this.system.removeComponent(this);
+    this.el.removeEventListener("grab-start", this._handleGrabStart);
+  },
+
+  beforeStep: function() {
+    if (this.el.body) {
+      const currentScale = this.el.object3D.scale.x;
+      const ratio = Math.min(1, (currentScale - this.initialScale) / (this.maxScale - this.initialScale));
```

Comment:
Couldnt this just be `THREE.MATH.clamp(currentScale/this.data.maxScale, 0, 1)`

Reply:
clamp can be used but initialScale still needs to be subtracted from both values before dividing or the math will be wrong